pkg/virt-controller/services/template.go

```diff
@@ -195,6 +199,56 @@ func setNodeAffinityForbiddenFeaturePolicy(vmi *v1.VirtualMachineInstance, pod *
 	}
 }
 
+func setNodeAffinityForCPUManager(pod *k8sv1.Pod) {
+	if pod.Spec.Affinity == nil {
+		pod.Spec.Affinity = &k8sv1.Affinity{}
+	}
+	if pod.Spec.Affinity.NodeAffinity == nil {
+		pod.Spec.Affinity.NodeAffinity = &k8sv1.NodeAffinity{}
+	}
+
+	if pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+		pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &k8sv1.NodeSelector{}
+	}
+
+	oldLabel := k8sv1.NodeSelectorRequirement{
+		Key:      v1.DeprecatedCPUManager,
+		Operator: k8sv1.NodeSelectorOpIn,
+		Values:   []string{"true"},
+	}
+	newLabel := k8sv1.NodeSelectorRequirement{
+		Key:      v1.CPUManager,
+		Operator: k8sv1.NodeSelectorOpIn,
+		Values:   []string{"true"},
+	}
+
+	if len(pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms) == 0 {
+		pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms =
+			[]k8sv1.NodeSelectorTerm{
+				{
+					MatchExpressions: []k8sv1.NodeSelectorRequirement{oldLabel},
+				},
+				{
+					MatchExpressions: []k8sv1.NodeSelectorRequirement{newLabel},
+				},
+			}
+	} else {
+		copies := []k8sv1.NodeSelectorTerm{}
+		for i := range pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms {
+			newTerm := pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[i].DeepCopy()
+			newTerm.MatchExpressions = append(newTerm.MatchExpressions, oldLabel)
+
+			pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[i].MatchExpressions = append(
+				pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[i].MatchExpressions, newLabel)
+			copies =
+				append(copies,
+					*newTerm,
+				)
+		}
+		pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = append(pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms, copies...)
```

Comment:
The idea is to add newLabel to the existing NodeSelectorTerm and to create a copy where you can add the oldLabel to? Reads a bit hard IMO.